app/models/location_comparable.rb

```diff
@@ -0,0 +1,22 @@
+class LocationComparable < Location
```

Comment:
Looks good. The main benefits of this approach are that this is simpler to work with outside of this class. In the new_attribute vs old_attribute case we had before, we were essentially forcing our caller to keep track of two objects' data and then pass those 6 attrs into the method we were calling before.